cdap-ui/app/features/admin/controllers/NamespaceDatasetMetadataController.js

```diff
@@ -1,15 +1,27 @@
 angular.module(PKG.name + '.feature.admin').controller('AdminNamespaceDatasetMetadataController',
```

Comment:
What issue does the change in this file fix?
Also, Why couldn't we do the equivalent simply by setting `$scope.metadata = null` in the case of a 404?